### PR TITLE
PGx tutorial notebooks

### PR DESCRIPTION
**Pull request by ahwagner**

### CODE REVIEW

**Comment by korikuzma** [comment]
@ahwagner VRS main does not have `GenotypeMember.copies`. It's now `GenotypeMember.count` as seen [here](https://github.com/ga4gh/vrs/blob/main/schema/vrs.yaml#L495). Either your VRS submodules need to be updated or we need to update the VRS main branch with this fix. This is all assuming you're using VRS main branch.

**Comment by korikuzma** [comment]
@ahwagner The above comment also applies to `Genotype` now having `count` rather than `copies`

**Comment by ahwagner** [comment]
The notebook was updated ahead of the demo to use changes on main, including `count`; where are you seeing `copies`?

**Comment by korikuzma** [comment]
@ahwagner only in the PGx - CYPC19 example 2!

**Comment by korikuzma** [comment]
Sorry, I should've specified the notebook name.

**Comment by ahwagner** [comment]
Ah, yeah. Thanks @korikuzma. That's a hold-over that didn't get updated. 

@wesleygoar the `PGx - CYPC19 example 2` notebook is your baby, want to update and confirm it all works, and then we can move the PR from draft to review-ready?

**Comment by korikuzma** [comment]
Ahh okay. Ya, this was me panicking having the local ready just in case the mybinder live demo didn’t work 😂 (so glad it did)

**Comment by korikuzma** [comment]
@wesleygoar You can make the PR ready for review if it's good to go

File: notebooks/PGx.ipynb [hunk 1]
```diff
@@ -0,0 +1,1180 @@
...
+    "from copy import deepcopy\n",
+    "\n",
+    "import re\n",
+    "import yaml\n",
+    "def ppo(o, indent=3):\n",
+    "    \"\"\"pretty print object as yaml\"\"\"\n",
+    "    print(yaml.dump(o.as_dict(), sort_keys=True, indent=indent))\n",
+    "    \n",
+    "from ga4gh.vrs.dataproxy import SeqRepoRESTDataProxy\n",
+    "from biocommons.seqrepo import SeqRepo\n",
```
Comment (korikuzma) [inline notebooks/PGx.ipynb:54]: Not used
```suggestion
    "from ga4gh.core import ga4gh_identify\n",
    "from jsonschema import validate\n",
    "import pathlib\n",
    "import csv\n",
    "from copy import deepcopy\n",
    "\n",
    "import re\n",
    "import yaml\n",
    "def ppo(o, indent=3):\n",
    "    \"\"\"pretty print object as yaml\"\"\"\n",
    "    print(yaml.dump(o.as_dict(), sort_keys=True, indent=indent))\n",
    "    \n",
    "from ga4gh.vrs.dataproxy import SeqRepoRESTDataProxy\n",
```

File: notebooks/PGx.ipynb [hunk 2]
```diff
@@ -0,0 +1,1180 @@
...
+    "def create_vrs_haplotype_from_genomic_definitions(genomic_vrs_defs, haploytpe_alt_defs):\n",
+    "    for haplotype_alt_def in haploytpe_alt_defs:\n",
+    "        allele_name = \"CYP2C19\" + haplotype_alt_def[0]\n",
+    "        special_definition = haplotype_alt_def[-1]\n",
+    "        allele_alts = haplotype_alt_def[1:-1]\n",
+    "        members = list()\n",
+    "        if special_definition:\n",
+    "            continue\n",
+    "        for base_allele, alt in zip(genomic_vrs_defs, allele_alts):\n",
+    "            definition_members = list()\n",
```
Comment (korikuzma) [inline notebooks/PGx.ipynb:183]: Not used
```suggestion
```

File: notebooks/PGx.ipynb [hunk 3]
```diff
@@ -0,0 +1,1180 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70e5856a-06d8-43a5-be5d-0ace27d07221",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "msg = {\n",
+    "    'type': 'StarAllele',\n",
+    "    'variation': h1_allele1.as_dict(),\n",
```
Comment (korikuzma) [inline notebooks/PGx.ipynb:1127]: `NameError: name 'h1_allele1' is not defined`

File: notebooks/PGx - CYP2C19 example 2.ipynb [hunk 4]
```diff
@@ -0,0 +1,516 @@
...
+    "from ga4gh.vrs import models, vrs_deref, vrs_enref\n",
+    "from ga4gh.core import ga4gh_identify, ga4gh_serialize, ga4gh_digest, ga4gh_deref\n",
+    "\n",
+    "import json\n",
+    "def ppo(o, indent=2):\n",
+    "    \"\"\"pretty print object as json\"\"\"\n",
+    "    print(json.dumps(o.as_dict(), sort_keys=True, indent=indent))\n",
+    "    \n",
+    "from ga4gh.vrs.dataproxy import SeqRepoRESTDataProxy\n",
+    "from biocommons.seqrepo import SeqRepo\n",
```
Comment (korikuzma) [inline notebooks/PGx - CYP2C19 example 2.ipynb:28]: Unused imports + why create function ppo when as_dict() does this for us? 
```suggestion
    "from ga4gh.vrs import models\n",
    "from ga4gh.core import ga4gh_identify\n",
    "\n",
    "import json\n",
    "from ga4gh.vrs.dataproxy import SeqRepoRESTDataProxy\n",
```

File: notebooks/PGx - CYP2C19 example 2.ipynb [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "9212b8c9-42ed-4ce3-b2e3-0d92f908f500",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#h1_allele1 = tlr.translate_from(hgvs_expression,'hgvs')"
+   ]
+  },
```
Comment (korikuzma) [inline notebooks/PGx - CYP2C19 example 2.ipynb:90]: We don't need commented out code blocks
```suggestion
```

File: notebooks/PGx - CYP2C19 example 2.ipynb [hunk 6]
```diff
@@ -0,0 +1,516 @@
...
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "h2"
```
Comment (korikuzma) [inline notebooks/PGx - CYP2C19 example 2.ipynb:151]: Makes it easier to read
```suggestion
    "h2.as_dict()"
```

File: notebooks/PGx - CYP2C19 example 2.ipynb [hunk 7]
```diff
@@ -0,0 +1,516 @@
...
+     },
+     "execution_count": 10,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "models.Number(value=1)"
+   ]
+  },
```
Comment (korikuzma) [inline notebooks/PGx - CYP2C19 example 2.ipynb:174]: Not really sure why we have this
```suggestion
```

File: notebooks/PGx - CYP2C19 example 2.ipynb [hunk 8]
```diff
@@ -0,0 +1,516 @@
...
+      "      }\n",
+      "    }\n",
+      "  ],\n",
+      "  \"type\": \"Genotype\"\n",
+      "}\n"
+     ]
+    }
+   ],
+   "source": [
+    "ppo(gt)"
```
Comment (korikuzma) [inline notebooks/PGx - CYP2C19 example 2.ipynb:330]:
```suggestion
    "gt.as_dict()"
```

File: notebooks/PGx - CYP2C19 example 2.ipynb [hunk 9]
```diff
@@ -0,0 +1,516 @@
...
```
Comment (korikuzma) [inline notebooks/PGx - CYP2C19 example 2.ipynb:478]:
```suggestion
    "gt.as_dict()"
```

**Review by korikuzma — CHANGES_REQUESTED** [review]

File: notebooks/Inlined and Referenced Objects.ipynb [hunk 10]
```diff
@@ -1,5 +1,15 @@
 {
  "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**NOTE:** A dynamic, web-based version of VRS-Python may be loaded through [mybinder.org](https://mybinder.org/v2/gh/ga4gh/vrs-python/pgxhttps://mybinder.org/v2/gh/ga4gh/vrs-python/pgx) where this notebook may be run without any local installation. Steps:\n",
```
[the remark this answers is not in the document]
Comment (ahwagner) [reply notebooks/Inlined and Referenced Objects.ipynb:7]: I think we should do both! This is an entry point for people that were following the ASHG tutorial and should stay (at least for the time being) for those that use that entry point.

Added issue #128 to address.

File: notebooks/PGx - CYP2C19 example 2.ipynb (hunk 4 above)
Comment (ahwagner) [reply notebooks/PGx - CYP2C19 example 2.ipynb]: I think as_dict helps, but ppo also adds some nice indent and sorting for display in the notebook.

File: notebooks/PGx.ipynb (hunk 3 above)
Comment (ahwagner) [reply notebooks/PGx.ipynb:2189]: Great catch! Added 17c1fad to fix.

File: notebooks/PGx.ipynb [hunk 11]
```diff
@@ -0,0 +1,2242 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "id": "a64936ee-9ca9-4979-83bd-7abd2b70e517",
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "SyntaxError",
```
Comment (korikuzma) [inline notebooks/PGx.ipynb:2104]: Should resolve syntax error

**Review by korikuzma — CHANGES_REQUESTED** [review]

**Review by korikuzma — APPROVED** [review]
🚀

**Comment by larrybabb** [comment]
@korikuzma very cool addition to the notebooks.

File: notebooks/Inlined and Referenced Objects.ipynb [hunk 12]
```diff
@@ -28,10 +38,7 @@
      "name": "stderr",
      "output_type": "stream",
      "text": [
-      "Removing allOf attribute from AbsoluteCopyNumber to avoid python-jsonschema-objects error.\n",
-      "Removing allOf attribute from SequenceInterval to avoid python-jsonschema-objects error.\n",
-      "Removing allOf attribute from RepeatedSequenceExpression to avoid python-jsonschema-objects error.\n",
-      "/Users/kxk102/Documents/ga4gh/vrs/venv/lib/python3.9/site-packages/python_jsonschema_objects/__init__.py:49: UserWarning: Schema version http://json-schema.org/draft-07/schema not recognized. Some keywords and features may not be supported.\n",
+      "/Users/ahw001/.local/share/virtualenvs/vrs-python-z7zdIrLG/lib/python3.9/site-packages/python_jsonschema_objects/__init__.py:49: UserWarning: Schema version http://json-schema.org/draft-07/schema not recognized. Some keywords and features may not be supported.\n",
```
[the remark this answers is not in the document]
Comment (ahwagner) [reply notebooks/Inlined and Referenced Objects.ipynb:41]: This is a python jsonschema objects warning that we don't control. It doesn't impact this notebook. Though I think the days of using the PJSO library are limited.

[the remark this answers is not in the document]
Comment (korikuzma) [reply notebooks/Inlined and Referenced Objects.ipynb:41]: https://github.com/cwacek/python-jsonschema-objects/issues/184

**Pull request merged**